src/AWS.Deploy.Orchestration/DeploymentManifestEngine.cs

```diff
@@ -0,0 +1,222 @@
+        /// If the manifest file does not exists then a new file is generated.
+        /// </summary>
+        public async Task UpdateLastDeployedStack(string stackName)
+        {
+            try
+            {
+                if (_session.AWSAccountId == null || _session.AWSRegion == null)
+                    throw new FailedToUpdateDeploymentManifestFileException("The AWS Account Id or Region is not defined.");
+
+                var deploymentManifestModel = await GetDeploymentManifest();
+                var lastDeployedStack = deploymentManifestModel?.LastDeployedStacks
+                    .FirstOrDefault(x => x.AWSAccountId.Equals(_session.AWSAccountId) && x.AWSRegion.Equals(_session.AWSRegion));
+
+                if (deploymentManifestModel != null && lastDeployedStack != null)
+                {
+                    if (lastDeployedStack != null)
+                    {
+                        lastDeployedStack.Stacks.Remove(stackName);
+                        lastDeployedStack.Stacks.Insert(0, stackName);
+                    }
+                    else
+                    {
+                        deploymentManifestModel.LastDeployedStacks.Add(new LastDeployedStack(
+                                _session.AWSAccountId,
+                                _session.AWSRegion,
+                                new List<string>() { stackName }));
+                    }
+                }
+                else
+                {
+
+                    var lastDeployedStacks = new List<LastDeployedStack> {
+                        new LastDeployedStack(
+                            _session.AWSAccountId,
+                            _session.AWSRegion,
+                            new List<string>() { stackName }) };
+                    deploymentManifestModel = new DeploymentManifestModel(lastDeployedStacks, new());
+                }
+
+                await WriteDeploymentManifestFile(deploymentManifestModel);
+            }
+            catch (Exception ex)
+            {
+                throw new FailedToUpdateDeploymentManifestFileException($"Failed to update the deployment manifest file " +
+                    $"to include the last deployed to stack '{stackName}'.", ex);
+            }
+        }
+
+        /// <summary>
+        /// This method updates the deployment manifest json file by deleting the stack that was most recently used.
+        /// </summary>
+        public async Task DeleteLastDeployedStack(string stackName)
+        {
+            try
+            {
+                var deploymentManifestModel = await GetDeploymentManifest();
+                var lastDeployedStack = deploymentManifestModel?.LastDeployedStacks
+                    .FirstOrDefault(x => x.AWSAccountId.Equals(_session.AWSAccountId) && x.AWSRegion.Equals(_session.AWSRegion));
+
+                if (deploymentManifestModel != null && lastDeployedStack != null)
```

Comment:
nit: lastDeployedStack has to be null if deploymentManifestModel so you could just check `lastDeployedStack != null`

Reply:
If i dont explicitly check deploymentManifestModel for null, I will get a Possible null reference error.